### DISCUSSION TITLE
Communication between two unconnected machines

### DISCUSSION THREAD

**Discussion by Herdismaria** [statelyai/xstate#1379] [General]
Hello, I'm relatively new to xState and I I am implementing a fairly complicated state machine in React.
A very short explanation: I have an ActiveTrips machine that can Add trips to it's context, end all trips and needs to poll a database for changes in the trips.
I then have a Trip machine that has Riding,Paused,Reserved,Ended states and can move between these. This machine also communicates with the server and ends and pauses trip.

I need a way for the Trip machine to communicate that it has ended to the ActiveTrips machine (and probably some other events).

I was going to use the Actor model for this but the useService hook does not take in config object like useMachine does and I need to send requests to our server and send different events based on the response and I don't know how to do this unless I get the send function from useMachine.

```const [current, send] = useMachine(tripMachine, {
    context: trip,
    services: {
      apiStartTrip: async () => {
        send('STARTED', { initialStatus: trip.initialStatus });
      },
      apiCanParkCheck: async context => {
        try {
          const parkStatus = await api.getActiveParkStatus(context.id);
          const canPark = parkStatus === null;

          send('PARKING_CHECK_DONE', {
            canPark,
            errorCode: parkStatus,
          });
        } catch (error) {
          const { code = null } = api.formatApiError(error);
          send('PARKING_CHECK_DONE', { canPark: false, errorCode: code });
        }
      },
```

So if I move this apiCanParkCheck inside services in the machine I can't use the send function. I also have services that call functions from React.Contexts that I can't move into the Machine definition object.

**Comment by davidkpiano** [comment]
Can you create a [CodeSandbox (XState template)](https://codesandbox.io/s/xstate-example-template-qcwed) with the current code that you have for this?

There are potentially different ways to model this. You can still use an Actor model pattern where the API communication is set up in the parent (`ActiveTrips`) machine, for instance. But it would be good to get a fuller idea of what you're trying to do with code.
